From 40a12694663610c5c471312178f19da33c8470ca Mon Sep 17 00:00:00 2001
From: Paul Gortmaker <paul.gortmaker@windriver.com>
Date: Fri, 11 Jul 2014 14:53:48 -0400
Subject: [PATCH] aufs: don't reference mutex->owner for RT_FULL

If RT_FULL is set, we source <linux/mutex_rt.h> instead of the
normal definitions for struct mutex in <linux/mutex.h> and the
RT variant doesn't have an owner field.

Ensure aufs doesn't go after ->owner when RT_FULL is enabled.

Signed-off-by: Paul Gortmaker <paul.gortmaker@windriver.com>
---
 fs/aufs/Kconfig | 1 -
 fs/aufs/i_op.c  | 2 ++
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/fs/aufs/Kconfig b/fs/aufs/Kconfig
index 7961c45215e5..c648d409814b 100644
--- a/fs/aufs/Kconfig
+++ b/fs/aufs/Kconfig
@@ -1,6 +1,5 @@
 config AUFS_FS
 	tristate "Aufs (Advanced multi layered unification filesystem) support"
-	depends on !PREEMPT_RT_BASE
 	help
 	Aufs is a stackable unification filesystem such as Unionfs,
 	which unifies several directories and provides a merged single
diff --git a/fs/aufs/i_op.c b/fs/aufs/i_op.c
index d82899afd40a..5291f57cbd05 100644
--- a/fs/aufs/i_op.c
+++ b/fs/aufs/i_op.c
@@ -426,9 +426,11 @@ out:
 
 void au_pin_hdir_set_owner(struct au_pin *p, struct task_struct *task)
 {
+#ifndef CONFIG_PREEMPT_RT_FULL
 #if defined(CONFIG_DEBUG_MUTEXES) || defined(CONFIG_SMP)
 	p->hdir->hi_inode->i_mutex.owner = task;
 #endif
+#endif
 }
 
 void au_pin_hdir_acquire_nest(struct au_pin *p)
-- 
2.0.1

